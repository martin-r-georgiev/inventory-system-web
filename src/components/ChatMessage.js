import React from 'react';

const ChatMessage = ({name, message, timestamp}) => {
    let formattedTime = new Date(timestamp).toLocaleTimeString();

    return (
        <p className="chatbox-message">
            <strong className="chatbox-username">{name}</strong>: <em>{message}</em>
            <p><i>{formattedTime}</i></p>
        </p>
    );
}

export default ChatMessage;