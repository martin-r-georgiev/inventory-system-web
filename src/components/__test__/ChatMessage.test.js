import React from 'react';
import ReactDOM from 'react-dom';
import { itTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

import ChatMessage from '../ChatMessage';

afterEach(cleanup);

it("Renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<ChatMessage/>, div);
});

it("Renders ChatMessage component correctly", () => {
    let timeDate = new Date(Date.UTC(2000, 1, 1));
    timeDate.setHours(0,0,0,0);
    let formattedTime = timeDate.toLocaleTimeString();

    const {getByTestId} = render(<ChatMessage name="User" message="content" timestamp={timeDate}/>);
    expect(getByTestId('chat-message')).toHaveTextContent("User");
    expect(getByTestId('chat-message')).toHaveTextContent("content");
    expect(getByTestId('chat-message')).toHaveTextContent(formattedTime);
});

it("Matches snapshot", () => {
    let timeDate = new Date(Date.UTC(2000, 1, 1));
    timeDate.setHours(0,0,0,0);
    const tree = renderer.create(<ChatMessage name="User" message="content" timestamp={timeDate}/>).toJSON();

    expect(tree).toMatchSnapshot();
});