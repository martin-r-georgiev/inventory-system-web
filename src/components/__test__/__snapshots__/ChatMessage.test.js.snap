// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="chatbox-message"
  data-testid="chat-message"
>
  <strong
    className="chatbox-username"
  >
    User
  </strong>
  : 
  <em>
    content
  </em>
  <p>
    <i>
      00:00:00
    </i>
  </p>
</div>
`;
